prism_compile.c

```diff
@@ -409,6 +409,53 @@ pm_static_literal_value(const pm_node_t *node, pm_scope_node_t *scope_node, pm_p
     }
 }
 
+/**
```

Comment:
I really like this in concept, but I'm a little confused by its place in this PR. More specifically, I think my order of preferences is:

1. Have this used everywhere that we're currently using dummy_line_node
2. Have this not exist
3. Have this used in some places, and not used in other places

It's clear this is better than the existing solution. However, having it used in some places and not others has ripple effects, like not allowing us to use many of the macros we've written to make prism compilation easier (`PM_POP`, `PM_PUTNIL`, `PM_DUP` etc). It then leaves us  with more things to update when we do, inevitably, move everything over to use this.

Reply:
Yeah, I'm sorry about putting that into this PR. The thing is, there were a lot of places where I was repeatedly calling out to determine the line number and it because really obvious in this PR.

My thinking here is that `dummy_line_node` is used everywhere, so it's not something we could fix in a single pull request. We're going to have to go incrementally with it. Right now the use of `pm_line_node_t` is isolated to pattern matching, so I was hoping we could start with it in this place.

The other thing is, all of those macros are going to have to change when we get line numbers correct. They're not going to be able to just reference `dummy_line_node`. So that work is going to have to be made anyway, so I figured I wouldn't dig us further down into using it with this PR.

I definitely understand your point and concern about the inconsistency here. I share it because I don't want to be doing things in two different ways. But I thought in this case it might be worth it since it's only used in pattern matching. What do you think?

Reply:
Totally agreed that the macros will have to change anyways, but with this PR it's not only that the macros will have to change, but also that we'll have to go through all of this code and convert to macros too, which is fine, just to me seemed less optimal than leaving in dummy line node generation here, including the macro usage in this commit, and then updating the macros just where they are defined when introducing this way of working with line numbers.

I think fixing dummy_line_node in one PR is going to be pretty doable, a search and replace for `dummy_line_node` itself, and then fixing places where we pass it / generate it.

That all said, this is definitely not a hill I'd die on, so if you prefer it like this, that's okay too, and we can update all this code to use the macros when we switch over